### DISCUSSION TITLE
Onboarding visualization tools out of the box in cQube

### DISCUSSION THREAD

**Discussion by sukhpreetssekhon** [Sunbird-cQube/community#370] [Ideas]
**Problem statement**
- cQube has its own custom visualization app. Adopters, who may already have their own visualization tools, have to add their visualisations to the cQube visualization app to extend their use cases or else be dependent on the cQube team to update / create new visualisations on the cQube visualization app.
 
**Way forward**
- Today, there are so many excellent open source visualization tools that are being used at scale. Can cQube not move away from its own visualization app and leverage existing open source visualization tools like Metabase / Superset and rather provide out of the box reference visualisations on these tools? 
- cQube can spend time on onboarding more visualization tools instead of focussing on creating / enabling new visualization types

**This will help** 
- adopters who are already using an existing visualization tool to quickly onboard to cube
- cQube team to get contribution of adapters from adopters of different visualization tools that can be linked to cQube

cc @surendrasinghs @rahul101001000 @chakshugautam

**Comment by surendrasinghs** [comment]
@vijayatgh @SanthoshVasabhaktula  What do you think?
@alok-os FYI

**Reply by vijayatgh** [reply]
There are 2 distinct mindsets we may have to distinguish.

(1) Simple and easy to adopt - Those who can derive value from out-of-the-box visualisation features. Right from ingestion to visualisation the tool is optimised for certain problem statements and ready to use. So this is a ready to use, out-of-the-box, solution approach.

(2) Customisation and plug-n-play capabilities - advanced features for various tweaks through configuration, extendability through various hooks and ability to plug-in point specific tools. This is a DIY solution approach.

What is the primary need in this case? 

Is it a OR or AND approach?
(1) OR (2) 
or
(1) AND (2)

Then, we will have to define sharply what's the core offering of the solution.

**Reply by sukhpreetssekhon** [reply]
We should be able to do both. 

Priority should be to enable users to get off the ground quickly and have visualisations out of the box. But these visualisations can be on an existing open source visualization apps.

Even if you do (1) what would be the argument of not using existing open source visualization apps as compared to building and maintaining your own?

**Reply by vijayatgh** [reply]
Technically either approach works. However, decision can be based on functional and non-functional (responsiveness, localisation etc) fitment analysis.

**Reply by SanthoshVasabhaktula** [reply]
In my view - cQube already supports open source visualization tools that can work on pre-computed csv/json datasets. 
@sukhpreetsamagra - So can you elaborate what we are looking for specifically here? Also what do you mean by on-boarding more visualization tools? aren't the tools supposed to work out of the box with datasets?

@surendrasinghs
